### ISSUE TITLE
Question on dealing with outliers after running BERTopic

### ISSUE THREAD

**Issue by cckimm** [MaartenGr/BERTopic#452]
Hello,

Thank you for the great package on Topic Modeling! This package is very helpful for my project.

I have a question about dealing with outliers (observations that fall under topic -1).

I tried to assign these observations manually using argmax(prob) but still am left with many  of them in outlier topic. 

I was wondering about your thoughts on running an additional BERTopic model on the set of outlier observations again to classify them in some sort of hybrid method.

Thank you.

**Comment by MaartenGr** [comment]
> I tried to assign these observations manually using argmax(prob) but still am left with many of them in outlier topic.

This should not be possible. The probability matrix actually does not include the outlier topic, all observations should be assigned to a non-outlier topic. Could you share the code for doing this?

**Comment by cckimm** [comment]
> > I tried to assign these observations manually using argmax(prob) but still am left with many of them in outlier topic.
> 
> This should not be possible. The probability matrix actually does not include the outlier topic, all observations should be assigned to a non-outlier topic. Could you share the code for doing this?

I have had an error to the code and I fixed it! Thank you.

But I have a separate question. If I save the topic model that I ran and load it back on, it does not save the classified results using the model on my set of documents right? I tried to include random seed in the model but since my tweet dataset has around 700,000 observation, the classification ends up in infinite loop every time I run it. Do you have possible solution for this problem?

Many thanks in advance!

**Comment by MaartenGr** [comment]
>  If I save the topic model that I ran and load it back on, it does not save the classified results using the model on my set of documents right?

If you mean the `topics` variables, then you can extract those after loading the model with:

```python
topics = topic_model._map_predictions(topic_model.hdbscan_model.labels_)
```

> I tried to include random seed in the model but since my tweet dataset has around 700,000 observation, the classification ends up in infinite loop every time I run it.

I am not entirely what you mean by this. When exactly does the model end up in an infinite loop? Is it during `fit_transform` the first time? Also, could you share your code for running BERTopic.

**Comment by cckimm** [comment]
> > If I save the topic model that I ran and load it back on, it does not save the classified results using the model on my set of documents right?
> 
> If you mean the `topics` variables, then you can extract those after loading the model with:
> 
> ```python
> topics = topic_model._map_predictions(topic_model.hdbscan_model.labels_)
> ```
> 
> > I tried to include random seed in the model but since my tweet dataset has around 700,000 observation, the classification ends up in infinite loop every time I run it.
> 
> I am not entirely what you mean by this. When exactly does the model end up in an infinite loop? Is it during `fit_transform` the first time? Also, could you share your code for running BERTopic.

The following is my code.
`topic_model3 = BERTopic(embedding_model = "all-MiniLM-L6-v2", calculate_probabilities=True, nr_topics="auto", n_gram_range=(1, 2), min_topic_size=100, verbose = True, low_memory = True)`

`tweets = posts['clean_original_tweet'].values
topics, probs = topic_model3.fit_transform(tweets)`

The following snap shot is the log that I am seeing.  My Twitter data contain around 1 million tweets.

![log1](https://user-images.githubusercontent.com/99993303/154832488-83004467-ee83-4db4-812a-77d8b6462a7e.png)

**Comment by MaartenGr** [comment]
From this, it does not necessarily mean that your code ends up in an infinite loop. It might simply take a while to perform the processing. Do note though that `calculate_probabilities=True` and ` low_memory=True`  might slow down the application significantly. If you have sufficient RAM, I would advise setting both to False as a start.
